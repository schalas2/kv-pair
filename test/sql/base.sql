CREATE EXTENSION pair;

      SELECT pair('foo', 'bar')
UNION SELECT pair('HEY'::text, 'bar')
UNION SELECT pair('foo'::text, 'woah'::text)
UNION SELECT pair('ick', 'foo'::text)
UNION SELECT pair('foo'::text, 1)
UNION SELECT pair(12.3, 'foo'::text)
UNION SELECT pair(1, 12)
ORDER BY pair;

      SELECT 'foo' ~> 'bar' AS arrowop
UNION SELECT 'HEY'::text ~> 'bar'
UNION SELECT 'foo'::text ~> 'woah'::text
UNION SELECT 'ick' ~> 'foo'::text
UNION SELECT 'foo'::text ~> 1
UNION SELECT 12.3 ~> 'foo'::text
UNION SELECT 1 ~> 12
ORDER BY arrowop;

CREATE TABLE kv (
    pair pair
);

INSERT INTO kv VALUES('foo' ~> 'bar'), (1 ~> 2);
SELECT (pair).k, (pair).v FROM kv ORDER BY (pair).k;

SELECT (pair('foo', 'bar')).k;
SELECT (pair('foo', 'bar')).v;

SELECT ('foo' ~> 'bar').k;
SELECT ('foo' ~> 'bar').v;
