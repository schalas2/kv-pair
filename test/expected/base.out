SET client_min_messages = warning;
CREATE EXTENSION pair;
RESET client_min_messages;
      SELECT pair('foo', 'bar')
UNION SELECT pair('HEY'::text, 'bar')
UNION SELECT pair('foo'::text, 'woah'::text)
UNION SELECT pair('ick', 'foo'::text)
UNION SELECT pair('foo'::text, 1)
UNION SELECT pair(12.3, 'foo'::text)
UNION SELECT pair(1, 12)
ORDER BY pair;
    pair    
------------
 (1,12)
 (12.3,foo)
 (HEY,bar)
 (foo,1)
 (foo,bar)
 (foo,woah)
 (ick,foo)
(7 rows)

      SELECT 'foo' ~> 'bar' AS arrowop
UNION SELECT 'HEY'::text ~> 'bar'
UNION SELECT 'foo'::text ~> 'woah'::text
UNION SELECT 'ick' ~> 'foo'::text
UNION SELECT 'foo'::text ~> 1
UNION SELECT 12.3 ~> 'foo'::text
UNION SELECT 1 ~> 12
ORDER BY arrowop;
  arrowop   
------------
 (1,12)
 (12.3,foo)
 (HEY,bar)
 (foo,1)
 (foo,bar)
 (foo,woah)
 (ick,foo)
(7 rows)

CREATE TABLE kv (
    pair pair
);
INSERT INTO kv VALUES('foo' ~> 'bar'), (1 ~> 2);
SELECT (pair).k, (pair).v FROM kv ORDER BY (pair).k;
  k  |  v  
-----+-----
 1   | 2
 foo | bar
(2 rows)

SELECT (pair('foo', 'bar')).k;
  k  
-----
 foo
(1 row)

SELECT (pair('foo', 'bar')).v;
  v  
-----
 bar
(1 row)

SELECT ('foo' ~> 'bar').k;
  k  
-----
 foo
(1 row)

SELECT ('foo' ~> 'bar').v;
  v  
-----
 bar
(1 row)

